File: 程序员面试金典/5.2二进制数转字符串/printBin.java
class Solution {
    public String printBin(double num) {
        if (num < 0 || num >= 1)
            return "ERROR";
        StringBuilder sb = new StringBuilder();
        sb.append("0.");
        while (num != 0) {
            if (sb.length() > 32)
                return "ERROR";
            num *= 2;
            if (num >= 1) {
                num -= 1;
                sb.append("1");
            }
            else
                sb.append("0");
        }
        return sb.toString();
    }
}

File: 程序员面试金典/5.2/printBin.py
class Solution:
    def printBin(self, num: float) -> str:
        ans = ['0', '.']
        for i in range(32):
            if num == 0:
                return ''.join(ans)
            num *= 2
            if num >= 1:
                num -= 1
                ans.append('1')
            else:
                ans.append('0')
        return 'ERROR'
